sdk/dotnet/Pulumi/Deployment/Deployment_Call.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright 2016-2021, Pulumi Corporation
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Google.Protobuf.WellKnownTypes;
+using Pulumi.Serialization;
+using Pulumirpc;
+
+namespace Pulumi
+{
+    public sealed partial class Deployment
+    {
+        void IDeployment.Call(string token, CallArgs args, Resource? self, CallOptions? options)
+            => Call<object>(token, args, self, options, convertResult: false);
+
+        Output<T> IDeployment.Call<T>(string token, CallArgs args, Resource? self, CallOptions? options)
+            => Call<T>(token, args, self, options, convertResult: true);
+
+        private Output<T> Call<T>(
+            string token, CallArgs args, Resource? self, CallOptions? options, bool convertResult)
+        {
+            var completionSource = new OutputCompletionSource<T>(null);
+            _runner.RegisterTask(
+                $"Call: token={token}",
+                CallAsync(token, args, self, options, completionSource, convertResult));
+            return completionSource.Output;
+        }
+
+        private async Task CallAsync<T>(
+            string token, CallArgs args, Resource? self, CallOptions? options,
+            OutputCompletionSource<T> completionSource, bool convertResult)
+        {
+            try
+            {
+                var (result, deps) = await CallRawAsync(token, args, self, options).ConfigureAwait(false);
+                if (!convertResult)
+                {
+                    completionSource.TrySetDefaultResult(isKnown: true);
```

Comment:
How do we know that known: true? Should unknown not flow through here?

Reply:
You're absolutely right. We should be flowing unknowns properly here. I screwed this up in the Python SDK (sigh; #7670), but Go and Node.js properly handle it. Will fix.